server/tests/integration_test.rs

```diff
@@ -0,0 +1,998 @@
+use std::borrow::Borrow;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::HashMap;
+use std::error::Error;
+use std::{env, io, time};
+
+use std::fs;
+use std::fs::canonicalize;
+use std::io::{BufRead, BufReader, Read, Write};
+use std::path;
+use std::path::Path;
+use std::process;
+use std::process::Child;
+use std::string::String;
+use std::sync::mpsc::channel;
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use crate::array::Array;
+use arrow_flight::flight_service_client::FlightServiceClient;
+use arrow_flight::utils::{flight_data_from_arrow_batch, flight_data_to_arrow_batch};
+use arrow_flight::{
+    Action, Criteria, FlightData, FlightDescriptor, FlightInfo, PutResult, SchemaAsIpc,
+};
+use assert_cmd::output::OutputError;
+use assert_cmd::Command;
+use datafusion::arrow::array::{
+    ArrayAccessor, Float32Array, Int64Array, PrimitiveArray, StringArray,
+    TimestampMicrosecondArray, TimestampMillisecondArray, UInt64Array,
+};
+use datafusion::arrow::datatypes::TimeUnit::Millisecond;
+use datafusion::arrow::datatypes::{DataType, Field, Schema, TimestampMillisecondType};
+use datafusion::arrow::error::ArrowError;
+use datafusion::arrow::ipc::convert::try_schema_from_ipc_buffer;
+use datafusion::arrow::record_batch::RecordBatch;
+use datafusion::arrow::{array, ipc};
+use futures::executor::block_on;
+use futures::stream;
+use log::{error, Log, Record};
+use prost::Message;
+use rand::Rng;
+use rusqlite::ffi::sqlite3_uint64;
+use serial_test::serial;
+use sqlparser::ast::DataType::Time;
+use sqlparser::test_utils::table;
+use sysinfo::{Pid, PidExt, ProcessExt, ProcessStatus, System, SystemExt};
+use tokio::io::AsyncBufReadExt;
+use tokio::runtime::Runtime;
+use tonic::transport::Channel;
+use tonic::{IntoStreamingRequest, Request};
+
+pub type ArrowValue = datafusion::arrow::datatypes::Float32Type;
+pub type ValueArray = PrimitiveArray<ArrowValue>;
+pub type TimeSeriesId = std::primitive::u64;
+pub type ArrowTimeSeriesId = datafusion::arrow::datatypes::UInt64Type;
+pub type ArrowTimestamp = TimestampMillisecondType;
+pub type Result<T> = std::result::Result<T, Box<dyn Error>>;
+
+#[test]
+#[serial]
+fn test_can_create_table() {
+    create_directory();
+
+    let mut flight_server = start_arrow_flight_server();
+
+    let table_name = "data".to_string();
+
+    if let Ok(rt) = Runtime::new() {
```

Comment:
I see your point, these have been removed.

Reply:
I believe that I had two reasons for this initially:

1. As you mention, I thought explicit error handling was the correct approach in order to include the internal error. But if you think that these are not necessary, and give us no advantage if a test fails then I will delete them. 
2. I was going to implement closing the process when the errors were triggered so that every other test did not fail. Since the server process is not ended, the other tests fail not because they actually trigger a failed case, but because another server is running in the background.